include/modem/pdn.h

```diff
@@ -183,29 +219,26 @@ int pdn_id_get(uint8_t cid);
 /**
  * @brief Retrieve dynamic parameters of a given PDP context.
  *
+ * @deprecated Use pdn_pdp_context_dynamic_params_get instead.
+ *
  * @param cid The PDP context ID.
  * @param[out] dns4_pri The address of the primary IPv4 DNS server. Optional, can be NULL.
  * @param[out] dns4_sec The address of the secondary IPv4 DNS server. Optional, can be NULL.
  * @param[out] ipv4_mtu The IPv4 MTU. Optional, can be NULL.
  *
  * @return Zero on success or an error code on failure.
  */
-int pdn_dynamic_params_get(uint8_t cid, struct in_addr *dns4_pri,
+__deprecated int pdn_dynamic_params_get(uint8_t cid, struct in_addr *dns4_pri,
 			   struct in_addr *dns4_sec, unsigned int *ipv4_mtu);
 
 /**
  * @brief Retrieve dynamic parameters of a given PDP context.
  *
- * @param cid The PDP context ID.
- * @param[out] dns6_pri The address of the primary IPv6 DNS server. Optional, can be NULL.
- * @param[out] dns6_sec The address of the secondary IPv6 DNS server. Optional, can be NULL.
- * @param[out] ipv6_mtu The IPv6 MTU. Optional, can be NULL.
+ * @param[in,out] populated_info PDN PDP context dynamic info.
  *
  * @return Zero on success or an error code on failure.
  */
-int pdn_dynamic_params_get_v6(uint8_t cid, struct in6_addr *dns6_pri,
-			   struct in6_addr *dns6_sec, unsigned int *ipv6_mtu);
-
+int pdn_pdp_context_dynamic_params_get(struct pdn_pdp_context_dynamic_info *populated_info);
```

Comment:
While it has been implemented like this in modem_shell, I think `cid` should be supplied to the function as a separate parameter to make the API clearer.
```suggestion
int pdn_context_dynamic_params_get(uint32_t cid, struct pdn_pdp_context_dynamic_info *populated_info);
```

Reply:
Ok, done.